pkg/kubelet/pleg/generic.go

```diff
@@ -264,6 +264,11 @@ func (g *GenericPLEG) relist() {
 		}
 		// Update the internal storage and send out the events.
 		g.podRecords.update(pid)
+
+		// Create containerExitCode map for lookup
+		containerExitCode := make(map[string]int)
+		containerDiedFlag := true
```

Comment:
why do we need `containerDiedFlag`? What if there are multiple containers that die in a pod? Will we get multiple events for them? As I understand, currently we will only log for the "first" container that died in a pod in such a case?

Reply:
containerDiedFlag is used for lazy filling up containerExitCode map. since there is no need to fill up it when no containerDied event happened at all.  if there is one containerDied event, containerExitCode will filled up, and containerDiedFlag becomes false (no more filling up needed if there are more died events)

For the second question: when multiple containers died in a pod, it will log out the event for each container (won't log out for already died containers)

For example: gke-metrics-agent pod has two containerDied event:
```
Mar 31 20:53:54 gke-kubelet-test-3-default-pool-fac27580-b2zb kubelet[1687]: I0331 20:53:54.382703    1687 kubelet.go:1984] SyncLoop (PLEG): "gke-metrics-agent-wsl75_kube-system(b1844c19-1179-4d43-a93d-8957a7168a1a)", event: &pleg.PodLifecycleEvent{ID:"b1844c19-1179-4d43-a93d-8957a7168a1a", Type:"ContainerDied", Data:"d81e6fdf07f0e822ab4208ddf608a6cff4ee15602c4c5dec3c3719298de067ea"}
Mar 31 20:53:55 gke-kubelet-test-3-default-pool-fac27580-b2zb kubelet[1687]: I0331 20:53:55.392376    1687 kubelet.go:1984] SyncLoop (PLEG): "gke-metrics-agent-wsl75_kube-system(b1844c19-1179-4d43-a93d-8957a7168a1a)", event: &pleg.PodLifecycleEvent{ID:"b1844c19-1179-4d43-a93d-8957a7168a1a", Type:"ContainerDied", Data:"f8330af7ea414c5a1a44a8306743fbf2eefa0f2e25bf0f0d7131c9abe914e9af"}
```

Reply:
As discussed offline, removed containerDiedFlag flag by checking length of containerExitCode